app/controllers/admin_controller.rb

```diff
@@ -160,8 +160,14 @@ def publish_comment
       if @comment.status == 1
         flash[:notice] = 'Comment already published.'
       else
+        first_timer_comment = (@comment.status == 4)
         @comment.publish
-        flash[:notice] = 'Comment published.'
+        @comment.author.unban
```

Comment:
What would happen when a user is already a normal user and not banned? Let me check if we need this. Other changes looks good.

Reply:
I think it doesn't changes anything. If you want, I can check the user's status and unban only when it's necessary. What do you think?

Reply:
Looks like you found fixed a bug :smile: . The user should be unbanned whenever his comment gets published but an alert may be generated extra saying user is unbanned if user is already normal too, I think we wouldn't see that now because currently moderators can only mark as spam from email but once we add publish buttons on comment then it would show the extra alert and may confuse moderators.

Reply:
So, how about adding condition on alert?

Reply:
@ValentinaTironi add condition to get it approved. Thanks!

Reply:
Sorry, I don't understand. You say show an alert saying `user is unbanned if user is already normal too` when `user.stats == 1`?

Reply:
@gauravano

Reply:
> Sorry, I don't understand. You say show an alert saying user is unbanned if user is already normal too when user.stats == 1?

Let's start again. You agree that an alert would be shown saying "User is unbanned, when moderator will approve the comment" ?

Reply:
@ValentinaTironi

Reply:
Yup, but I think it has to be showed when user is banned

Reply:
@gauravano

Reply:
And, in case user is not banned? Do we still need to show that alert? @ValentinaTironi

Reply:
Nope

Reply:
So, I do the condition. `if user.status == 0` and if it's true `user.unban and show the alert`. Am I right?

Reply:
@gauravano

Reply:
But, according to current code 
![snippet_admin_unban](https://user-images.githubusercontent.com/20878070/49743320-7dd79180-fcc0-11e8-97d6-225e81d70c0b.png)

alert would be shown.

Reply:
So, let's add condition to call unban only when needed. In case, a comment is approved from dashboard we don't need to call unban. Make sense?

Reply:
That method is from `admin_controller`, so isn't that using here:
<img width="420" alt="screen shot 2018-12-10 at 12 49 27 pm" src="https://user-images.githubusercontent.com/20969831/49743683-126ad100-fc7a-11e8-9ac5-87472012bc60.png">
In the controller is using the `unban` method defines in `drupal_user` model:
<img width="238" alt="screen shot 2018-12-10 at 12 50 17 pm" src="https://user-images.githubusercontent.com/20969831/49743735-2b738200-fc7a-11e8-8dd2-20900d538d10.png">
So, we need the alert how I showed in the first image

Reply:
ok, now I am confused.

Reply:
I have to see the code.

Reply:
But in first place, why are we calling the unban method if user is normal already?

Reply:
I don't know, I got confused I think

Reply:
Please, let me know if you are agree with me in this comment https://github.com/publiclab/plots2/pull/4217#discussion_r240265441 and I'm gonna do the pull request :) Hope we can finish this before claiming tasks time finish

Reply:
Let's add a condition something like this - 
If user is banned, unban the user. What do you think?